### ISSUE TITLE
How to build with 32-bit compatibility??

### ISSUE THREAD

**Issue by VajiraPrabuddhaka** [mongodb/mongo-php-driver#671]
In my pc I have already installed lampp. So when I build mongodb.so using [this](http://php.net/manual/en/mongodb.installation.manual.php) instructions, lampp suddenly says 

> "XAMPP is currently only availably as 32 bit application. Please use a 32 bit compatibility library for your system"
Can anyone help me to build this with 32-bit compatibility as it suggests??

**Comment by derickr** [comment]
It seems to e some generic error if PHP can't start. (https://shrinathsuccess.wordpress.com/tag/xampp-is-currently-only-availably-as-32-bit-application-please-use-a-32-bit-compatibility-library-for-your-system/) What does the following output:

```/opt/lampp/bin/php -v```

There is no problem building the MongoDB extension with a 32-bit PHP at all, but it *is* possible that you have to run "export CFLAGS=-m32" before running ```./configure```

**Comment by VajiraPrabuddhaka** [comment]
Here is the output for above 

> PHP Warning:  PHP Startup: Unable to load dynamic library '/opt/lampp/lib/php/extensions/no-debug-non-zts-20160303/mongodb.so' - /opt/lampp/lib/php/extensions/no-debug-non-zts-20160303/mongodb.so: wrong ELF class: ELFCLASS32 in Unknown on line 0
Warning: PHP Startup: Unable to load dynamic library '/opt/lampp/lib/php/extensions/no-debug-non-zts-20160303/mongodb.so' - /opt/lampp/lib/php/extensions/no-debug-non-zts-20160303/mongodb.so: wrong ELF class: ELFCLASS32 in Unknown on line 0
PHP 7.1.9 (cli) (built: Sep 18 2017 02:29:39) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Xdebug v2.5.5, Copyright (c) 2002-2017, by Derick Rethans

And I also done the `CFLAGS=-m32` and it successfully fixed above error, but still it is not working

**Comment by derickr** [comment]
Heh. That is odd. It indicates it wants a 64-bit binary (ELFCLASS64) and not the ELFCLASS32 that you just built. Can you show the output of the following commands:

```file /opt/lampp/bin/php```
```file /opt/lampp/lib/php/extensions/no-debug-non-zts-20160303/mongodb.so```

**Comment by VajiraPrabuddhaka** [comment]
Now I changed to 64-bit and still getting initial error as in the first comment,

Followings are the outputs for titled commands:

`file /opt/lampp/bin/php`

> /opt/lampp/bin/php: symbolic link to php-7.1.9

`file /opt/lampp/lib/php/extensions/no-debug-non-zts-20160303/mongodb.so`

> /opt/lampp/lib/php/extensions/no-debug-non-zts-20160303/mongodb.so: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=b71d583b92c4218566d5711e927c0bb93b14dc4f, with debug_info, not stripped

`/opt/lampp/bin/php -v`

> /opt/lampp/bin/php: relocation error: /opt/lampp/lib/php/extensions/no-debug-non-zts-20160303/mongodb.so: symbol SSL_library_init, version OPENSSL_1.0.0 not defined in file libssl.so.1.0.0 with link time reference

**Comment by jmikola** [comment]
> file /opt/lampp/bin/php
>
> `/opt/lampp/bin/php: symbolic link to php-7.1.9`

Can you step through the symlink and run `file` on the actual binary it points to?

**Comment by VajiraPrabuddhaka** [comment]
@jmikola it is pointed to `php-7.1.9` binary file in the same directory, here is the details
Command: `file  /opt/lampp/bin/php-7.1.9` 
Output:

> php-7.1.9: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.9, stripped

>

**Comment by flyneko** [comment]
@VajiraPrabuddhaka Have you fixed this issue? I have the same error

**Comment by VajiraPrabuddhaka** [comment]
@flyneko No, I couldn't fix it, instead I turned to use global` apache` server installed in l`inux` with global `PHP` installation for `linux`, rather than using `lampp`
